internal/syft/syft.go

```diff
@@ -51,18 +56,53 @@ func (s *Syft) ExecuteSyft(img kubernetes.ImageDigest) string {
 		return filePath
 	}
 
-	cmd := exec.Command("syft", imagePath, "-o", s.SbomFormat)
-	var errb bytes.Buffer
-	cmd.Stderr = &errb
-	stdout, err := cmd.Output()
+	src, cleanup, err := source.New(filepath.Join("oci-archive:", imagePath), nil, nil)
+	if err != nil {
+		panic(fmt.Errorf("failed to construct source from input %s: %w", imagePath, err))
+	}
+	if cleanup != nil {
+		defer cleanup()
+	}
 
-	os.RemoveAll(workDir)
+	bi, ok := debug.ReadBuildInfo()
+	if !ok {
+		logrus.Warnf("failed to read build info")
+	}
+
+	descriptor := sbom.Descriptor{
+		Name: "syft",
+	}
+
+	for _, dep := range bi.Deps {
+		if strings.EqualFold("github.com/anchore/syft", dep.Path) {
+			descriptor.Version = dep.Version
+		}
+	}
 
+	result := sbom.SBOM{
+		Source: src.Metadata,
+		// TODO: we should have helper functions for getting this built from exported library functions
+	}
+
+	c := cataloger.DefaultConfig()
+	c.Search.Scope = source.SquashedScope
+	packageCatalog, relationships, theDistro, err := syft.CatalogPackages(src, c)
 	if err != nil {
-		logrus.WithError(err).WithField("stderr", errb.String()).Error("Syft stopped with error")
-		return filePath
+		panic(err)
 	}
 
+	result.Artifacts.PackageCatalog = packageCatalog
+	result.Artifacts.LinuxDistribution = theDistro
+	result.Relationships = relationships
+
+	// you can use other formats such as format.CycloneDxJSONOption or format.SPDXJSONOption ...
+	b, err := syft.Encode(result, format.JSONOption)
```

Comment:
We have to integrate the configured format (`s.SbomFormat` here again).

Reply:
Thank you so much, I missed that option 🙋🏻‍♂️ It should be okay right now, can you please check again ✌️